src/App.tsx

```diff
@@ -15,48 +23,92 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+const getPreparedGoods = (
+  goods: string[],
+  sortedBy: SortBy,
+  isReversed: boolean,
+): string[] => {
+  let preparedGoods = [...goods];
+
+  if (sortedBy) {
+    preparedGoods.sort((goodA, goodB) => {
+      switch (sortedBy) {
+        case SortBy.name:
+          return goodA.localeCompare(goodB);
+
+        case SortBy.length:
+          return goodA.length - goodB.length;
+
+        case SortBy.default:
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    preparedGoods = preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+};
+
 export const App: React.FC = () => {
+  const [sortBy, setSortBy] = useState(SortBy.default);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const reset = () => {
+    setSortBy(SortBy.default);
+    setIsReversed(false);
+  };
+
+  const goods = getPreparedGoods(goodsFromServer, sortBy, isReversed);
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={cn('button is-info', {
+            'is-light': sortBy !== SortBy.name,
+          })}
+          onClick={() => setSortBy(SortBy.name)}
         >
           Sort alphabetically
         </button>
-
         <button
           type="button"
-          className="button is-success is-light"
+          className={cn('button is-success', {
+            'is-light': sortBy !== SortBy.length,
+          })}
+          onClick={() => setSortBy(SortBy.length)}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={cn('button is-warning', {
+            'is-light': !isReversed,
+          })}
+          onClick={() => setIsReversed(!isReversed)}
         >
           Reverse
         </button>
 
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
+        {(sortBy !== SortBy.default || isReversed) && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => reset()}
```

Comment:
```suggestion
            onClick={reset}
```
can be more simplify